From 3ce4f13705b6d5c0381d7cf156580327597fd65b Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Tue, 13 Jan 2026 14:53:52 +0100
Subject: [PATCH] ksystemclipboard: Check m_thread in destructor

The thread might be null

BUG: 514512
(cherry picked from commit 9cf68eef2543afebc74d2a87aee1d49ef392c11c)
---
 src/systemclipboard/waylandclipboard.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/systemclipboard/waylandclipboard.cpp b/src/systemclipboard/waylandclipboard.cpp
index 74872c3..b90bc3c 100644
--- a/src/systemclipboard/waylandclipboard.cpp
+++ b/src/systemclipboard/waylandclipboard.cpp
@@ -619,7 +619,7 @@ WaylandClipboard::WaylandClipboard(QObject *parent)
 
 WaylandClipboard::~WaylandClipboard()
 {
-    if (m_thread->isRunning()) {
+    if (m_thread && m_thread->isRunning()) {
         m_thread->syncQueue();
         m_thread->wait();
     }
-- 
GitLab

